src/commands/new/glee.ts

```diff
@@ -1,55 +1,139 @@
-        break;
-      case 'EACCES':
-        this.error(`Unable to create the project. We tried to access the "${PROJECT_DIRECTORY}" directory but it was not possible due to file access permissions. Please check the write permissions of your current working directory ("${process.cwd()}").`);
-        break;
-      case 'EPERM':
-        this.error(`Unable to create the project. We tried to create the "${PROJECT_DIRECTORY}" directory but the operation requires elevated privileges. Please check the privileges for your current user.`);
-        break;
-      default:
-        this.error(`Unable to create the project. Please check the following message for further info about the error:\n\n${err}`);
-      }
+
+    const GLEE_TEMPLATES_DIRECTORY = resolve(
+      __dirname,
+      '../../../assets/create-glee-app/templates/',
+      templateName
+    );
+
+    const currentTemplateName =
+      'https://github.com/KhudaDad414/glee-generator-template';
+
+    if (file && templateName && templateName !== 'default') {
+      this.error('You cannot use both --t and --f in the same command.');
     }
+    if (file) {
+      try {
+        const asyncapiInput = (await load(file)) || (await load());
+
+        const serversObject = asyncapiInput.toJson().servers;
+        const servers = Object.keys(serversObject);
+        const remoteServers = [];
+        for (const server of servers) {
+          const isRemote = await CliUx.ux.confirm(
+            `Is "${server}" a remote server`
+          );
+          remoteServers.push({ server, isRemote });
+        }
+
+        const selectedRemoteServers = remoteServers
+          .filter(({ isRemote }) => isRemote)
+          .map(({ server }) => server);
+        const asyncapiObject = asyncapiInput.toJson();
+        asyncapiObject['x-remoteServers'] = selectedRemoteServers;
+
+        delete asyncapiObject.filePath;
+        delete asyncapiObject.kind;
 
-    try {
-      await fs.copy(GLEE_TEMPLATES_DIRECTORY, PROJECT_DIRECTORY);
-      await fPromises.rename(`${PROJECT_DIRECTORY}/env`, `${PROJECT_DIRECTORY}/.env`);
-      await fPromises.rename(`${PROJECT_DIRECTORY}/gitignore`, `${PROJECT_DIRECTORY}/.gitignore`);
-      await fPromises.rename(`${PROJECT_DIRECTORY}/README-template.md`, `${PROJECT_DIRECTORY}/README.md`);
-      this.log(`Your project "${projectName}" has been created successfully!\n\nNext steps:\n\n  cd ${projectName}\n  npm install\n  npm run dev\n\nAlso, you can already open the project in your favorite editor and start tweaking it.`);
-    } catch (err) {
-      this.error(`Unable to create the project. Please check the following message for further info about the error:\n\n${err}`);
+        const updatedAsyncApiContent = yaml.dump(asyncapiObject, {
+          lineWidth: -1,
+        });
+
+        const currentFileDirectory = path.join(__dirname, '../../..', file);
```

Comment:
we do not want to save the changes back to the file. I don't expect my file to be modified after this operation. I think it would be best to create and temporary file, generate a glee project from it, and then remove it.

Reply:
created the temporary file.